File: ExamPreparation/ShopingList/C# Solution/ShoppingList/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.Web.Mvc;

namespace ShoppingList.Models
{
    public class Product
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MinLength(1)]
        public string Name { get; set; }

        [Required]
        [MinLength(1)]
        public int Priority { get; set; }

        [Required]
        [MinLength(1)]
        public int Quantity { get; set; }

        [Required]
        [MinLength(1)]
        public string Status { get; set; }
    }
}

File: ExamPreparation/ShopingList/JavaScript Solution/models/Product.js
const mongoose = require('mongoose');

let productSchema = mongoose.Schema({
    name: {type: 'string', required: 'true'},
    priority: {type: 'number', required: 'true'},
    quantity: {type: 'number', required: 'true'},
    status: {type: 'string', required: 'true'}
});

let Product = mongoose.model('Product', productSchema);

module.exports = Product;